test_conformance/subgroups/test_subgroup_ballot.cpp

```diff
@@ -78,25 +126,54 @@ template <typename Ty> struct BALLOT
                 {
                     current_sbs = wg_offset + sbs > lws ? lws - wg_offset : sbs;
                 }
-                for (wi_id = 0; wi_id < current_sbs; ++wi_id)
+
+                bs128 expected_result_bs = 0;
+
+                std::set<int> active_work_items;
```

Comment:
I would suggest moving this outside of all loops, and use .clear() here.
However, this would be a minor performance tune, and ultimately I don't think it ought to block the commit.

Reply:
How expensive do you think the bs128 use would be in comparison?

Reply:
Sorry, github gives more context than I intended. I meant strictly the std::set, and nothing else.
My reasoning was that clear() might recycle previous heap allocations.

Reply:
Right, but might avoiding having bs128s falling in and out of scope, too, also recycle previous heap allocations? I wonder if either might be premature optimization, though.

Reply:
I don't know what bs128 is, and it seems unrelated to std::set (from my cursory view).
This is fine as-is. I merely wanted to provide some feedback on this line and am not expecting any changes :)
Ultimately whatever gains this may have might be immesurable when compared to the time consumed by the rest of the function.

Reply:
The bs128 is a 128-bit [std::bitset](https://en.cppreference.com/w/cpp/utility/bitset). I expect it will involve some heap use.

Reply:
Perhaps worth profiling this, but there's a fair chance that this code will be changing quite a bit anyway.